core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -154,8 +154,21 @@ public void start() throws IOException, IllegalStateException {
 
                     frameMat.copyTo(frameOutputSocket.getValue().get());
                     frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                    long thisMoment = System.currentTimeMillis();
-                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+
+                    final long thisMoment = System.nanoTime();
+                    final long elapsedTime = thisMoment - lastFrame;
+                    if (elapsedTime == 0) {
+                        /* This can happen when using system.currentTimeMillis():
+                         * https://github.com/WPIRoboticsProjects/GRIP/issues/199
+                         * As such we use system.nanoTime().
+                         * However, we add an assertion to ensure this bug doesn't reappear.
+                         */
+                        assert elapsedTime != 0 : "The time between grabbing two frames was less than 1 nanosecond";
```

Comment:
So I looked up using assertions in production code. Apparently this is an ongoing debate, but most people seem to agree with Tom that it should be avoided- instead of an assert, you should be handling errors explicitly- IE If you want to keep the test in, you should have something to handle it when it fails and you should be avoiding using "assert" in general. If you'd like to test this condition, re-writing it so that you don't use assert and the error is handled would be OK with me. But I think the easier option would be to simply take it out....